panic: DateTime was not found

### Expected Behaviour
Can generate code and resolver from schema.graphql
### Actual Behavior
```
panic: DateTime was not found

goroutine 1 [running]:
ws/gqlqs/vendor/github.com/99designs/gqlgen/codegen.(*builder).bindField.func1(0xc0009a1d40, 0xc0007c7960)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/codegen/field.go:81 +0xac
ws/gqlqs/vendor/github.com/99designs/gqlgen/codegen.(*builder).bindField(0xc0007c7960, 0xc00043c800, 0xc0009a1d40, 0x16cc180, 0xc000455ee0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/codegen/field.go:168 +0xbd6
ws/gqlqs/vendor/github.com/99designs/gqlgen/codegen.(*builder).buildField(0xc0007c7960, 0xc00043c800, 0xc00019d7a0, 0xc0009a1c00, 0x0, 0x0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/codegen/field.go:64 +0x31d
ws/gqlqs/vendor/github.com/99designs/gqlgen/codegen.(*builder).buildObject(0xc0007c7960, 0xc0000e7c80, 0x15ef4c0, 0x6, 0x0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/codegen/object.go:72 +0x4d4
ws/gqlqs/vendor/github.com/99designs/gqlgen/codegen.BuildData(0xc0001e40c0, 0x16cca00, 0x1acd4a0, 0x0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/codegen/data.go:76 +0x395
ws/gqlqs/vendor/github.com/99designs/gqlgen/api.Generate(0xc0001e40c0, 0x0, 0x0, 0x0, 0x0, 0x162fef8)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/api/generate.go:37 +0x22d
ws/gqlqs/vendor/github.com/99designs/gqlgen/cmd.glob..func1(0xc0001ec000)
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/cmd/gen.go:39 +0xa2
ws/gqlqs/vendor/github.com/urfave/cli.HandleAction(0x152f7a0, 0x16302f8, 0xc0001ec000, 0xc0001163c0, 0x0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/urfave/cli/app.go:492 +0x7c
ws/gqlqs/vendor/github.com/urfave/cli.(*App).Run(0xc0001e0000, 0xc000020020, 0x1, 0x1, 0x0, 0x0)
	/Users/gtm/go/src/ws/gland/vendor/github.com/urfave/cli/app.go:264 +0x57c
ws/gqlqs/vendor/github.com/99designs/gqlgen/cmd.Execute()
	/Users/gtm/go/src/ws/gland/vendor/github.com/99designs/gqlgen/cmd/root.go:40 +0x224
main.main()
	/Users/gtm/go/src/ws/gland/scripts/gqlgen.go:8 +0x20
exit status 2
```
### Minimal graphql.schema and models to reproduce
```
type Query {
  countries: [Country!]!
}

type Mutation {
  createCountry(input: CountryCreateInput!): Country
}

# input
input CountryCreateInput {
  name: String!
  code: String!
}

scalar DateTime
scalar Json

type Country {
  name: String!
  code: String!
  images: [Json!]!
  createdAt: DateTime!
}

```

--- a/codegen/testserver/generated.go
+++ b/codegen/testserver/generated.go
@@ -132,6 +132,7 @@ type ComplexityRoot struct {
 		Autobind               func(childComplexity int) int
 		DeprecatedField        func(childComplexity int) int
 		Panics                 func(childComplexity int) int
+		DefaultScalar          func(childComplexity int, arg string) int
 		ValidType              func(childComplexity int) int
 	}
 
@@ -194,6 +195,7 @@ type QueryResolver interface {
 	Autobind(ctx context.Context) (*Autobind, error)
 	DeprecatedField(ctx context.Context) (string, error)
 	Panics(ctx context.Context) (*Panics, error)
+	DefaultScalar(ctx context.Context, arg string) (string, error)
 	ValidType(ctx context.Context) (*ValidType, error)
 }
 type SubscriptionResolver interface {
@@ -594,6 +596,18 @@ func (e *executableSchema) Complexity(typeName, field string, childComplexity in
 
 		return e.complexity.Query.Panics(childComplexity), true
 
+	case "Query.DefaultScalar":
+		if e.complexity.Query.DefaultScalar == nil {
+			break
+		}
+
+		args, err := ec.field_Query_defaultScalar_args(context.TODO(), rawArgs)
+		if err != nil {
+			return 0, false
+		}
+
+		return e.complexity.Query.DefaultScalar(childComplexity, args["arg"].(string)), true
+
 	case "Query.ValidType":
 		if e.complexity.Query.ValidType == nil {
 			break
@@ -838,6 +852,13 @@ type Panics {
 }
 
 scalar MarshalPanic
+`},
+	&ast.Source{Name: "scalar_default.graphql", Input: `extend type Query {
+    defaultScalar(arg: DefaultScalarImplementation! = "default"): DefaultScalarImplementation!
+}
+
+""" This doesnt have an implementation in the typemap, so it should act like a string """
+scalar DefaultScalarImplementation
 `},
 	&ast.Source{Name: "schema.graphql", Input: `type Query {
     invalidIdentifier: InvalidIdentifier
@@ -1133,6 +1154,20 @@ func (ec *executionContext) field_Query___type_args(ctx context.Context, rawArgs
 		}
 	}
 	args["name"] = arg0
+	return args, nil
+}
+
+func (ec *executionContext) field_Query_defaultScalar_args(ctx context.Context, rawArgs map[string]interface{}) (map[string]interface{}, error) {
+	var err error
+	args := map[string]interface{}{}
+	var arg0 string
+	if tmp, ok := rawArgs["arg"]; ok {
+		arg0, err = ec.unmarshalNDefaultScalarImplementation2string(ctx, tmp)
+		if err != nil {
+			return nil, err
+		}
+	}
+	args["arg"] = arg0
 	return args, nil
 }
 
@@ -2766,6 +2801,39 @@ func (ec *executionContext) _Query_panics(ctx context.Context, field graphql.Col
 	rctx.Result = res
 	ctx = ec.Tracer.StartFieldChildExecution(ctx)
 	return ec.marshalOPanics2ᚖgithubᚗcomᚋ99designsᚋgqlgenᚋcodegenᚋtestserverᚐPanics(ctx, field.Selections, res)
+}
+
+func (ec *executionContext) _Query_defaultScalar(ctx context.Context, field graphql.CollectedField) graphql.Marshaler {
+	ctx = ec.Tracer.StartFieldExecution(ctx, field)
+	defer func() { ec.Tracer.EndFieldExecution(ctx) }()
+	rctx := &graphql.ResolverContext{
+		Object: "Query",
+		Field:  field,
+		Args:   nil,
+	}
+	ctx = graphql.WithResolverContext(ctx, rctx)
+	rawArgs := field.ArgumentMap(ec.Variables)
+	args, err := ec.field_Query_defaultScalar_args(ctx, rawArgs)
+	if err != nil {
+		ec.Error(ctx, err)
+		return graphql.Null
+	}
+	rctx.Args = args
+	ctx = ec.Tracer.StartFieldResolverExecution(ctx, rctx)
+	resTmp := ec.FieldMiddleware(ctx, nil, func(rctx context.Context) (interface{}, error) {
+		ctx = rctx // use context from middleware stack in children
+		return ec.resolvers.Query().DefaultScalar(rctx, args["arg"].(string))
+	})
+	if resTmp == nil {
+		if !ec.HasError(rctx) {
+			ec.Errorf(ctx, "must not be null")
+		}
+		return graphql.Null
+	}
+	res := resTmp.(string)
+	rctx.Result = res
+	ctx = ec.Tracer.StartFieldChildExecution(ctx)
+	return ec.marshalNDefaultScalarImplementation2string(ctx, field.Selections, res)
 }
 
 func (ec *executionContext) _Query_validType(ctx context.Context, field graphql.CollectedField) graphql.Marshaler {
@@ -4993,6 +5061,20 @@ func (ec *executionContext) _Query(ctx context.Context, sel ast.SelectionSet) gr
 				res = ec._Query_panics(ctx, field)
 				return res
 			})
+		case "defaultScalar":
+			field := field
+			out.Concurrently(i, func() (res graphql.Marshaler) {
+				defer func() {
+					if r := recover(); r != nil {
+						ec.Error(ctx, ec.Recover(ctx, r))
+					}
+				}()
+				res = ec._Query_defaultScalar(ctx, field)
+				if res == graphql.Null {
+					invalid = true
+				}
+				return res
+			})
 		case "validType":
 			field := field
 			out.Concurrently(i, func() (res graphql.Marshaler) {
@@ -5410,6 +5492,14 @@ func (ec *executionContext) unmarshalNBoolean2bool(ctx context.Context, v interf
 
 func (ec *executionContext) marshalNBoolean2bool(ctx context.Context, sel ast.SelectionSet, v bool) graphql.Marshaler {
 	return graphql.MarshalBoolean(v)
+}
+
+func (ec *executionContext) unmarshalNDefaultScalarImplementation2string(ctx context.Context, v interface{}) (string, error) {
+	return graphql.UnmarshalString(v)
+}
+
+func (ec *executionContext) marshalNDefaultScalarImplementation2string(ctx context.Context, sel ast.SelectionSet, v string) graphql.Marshaler {
+	return graphql.MarshalString(v)
 }
 
 func (ec *executionContext) unmarshalNID2int(ctx context.Context, v interface{}) (int, error) {

--- a/codegen/testserver/resolver.go
+++ b/codegen/testserver/resolver.go
@@ -116,6 +116,9 @@ func (r *queryResolver) DeprecatedField(ctx context.Context) (string, error) {
 func (r *queryResolver) Panics(ctx context.Context) (*Panics, error) {
 	panic("not implemented")
 }
+func (r *queryResolver) DefaultScalar(ctx context.Context, arg string) (string, error) {
+	panic("not implemented")
+}
 func (r *queryResolver) ValidType(ctx context.Context) (*ValidType, error) {
 	panic("not implemented")
 }

--- a/codegen/testserver/scalar_default.graphql
+++ b/codegen/testserver/scalar_default.graphql
@@ -1,0 +1,6 @@
+extend type Query {
+    defaultScalar(arg: DefaultScalarImplementation! = "default"): DefaultScalarImplementation!
+}
+
+""" This doesnt have an implementation in the typemap, so it should act like a string """
+scalar DefaultScalarImplementation

--- a/codegen/testserver/scalar_default_test.go
+++ b/codegen/testserver/scalar_default_test.go
@@ -1,0 +1,34 @@
+package testserver
+
+import (
+	"context"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/99designs/gqlgen/client"
+	"github.com/99designs/gqlgen/handler"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDefaultScalarImplementation(t *testing.T) {
+	resolvers := &Stub{}
+
+	srv := httptest.NewServer(handler.GraphQL(NewExecutableSchema(Config{Resolvers: resolvers})))
+	c := client.New(srv.URL)
+
+	resolvers.QueryResolver.DefaultScalar = func(ctx context.Context, arg string) (i string, e error) {
+		return arg, nil
+	}
+
+	t.Run("with arg value", func(t *testing.T) {
+		var resp struct{ DefaultScalar string }
+		c.MustPost(`query { defaultScalar(arg: "fff") }`, &resp)
+		require.Equal(t, "fff", resp.DefaultScalar)
+	})
+
+	t.Run("with default value", func(t *testing.T) {
+		var resp struct{ DefaultScalar string }
+		c.MustPost(`query { defaultScalar  }`, &resp)
+		require.Equal(t, "default", resp.DefaultScalar)
+	})
+}

--- a/codegen/testserver/stub.go
+++ b/codegen/testserver/stub.go
@@ -42,6 +42,7 @@ type Stub struct {
 		Autobind               func(ctx context.Context) (*Autobind, error)
 		DeprecatedField        func(ctx context.Context) (string, error)
 		Panics                 func(ctx context.Context) (*Panics, error)
+		DefaultScalar          func(ctx context.Context, arg string) (string, error)
 		ValidType              func(ctx context.Context) (*ValidType, error)
 	}
 	SubscriptionResolver struct {
@@ -157,6 +158,9 @@ func (r *stubQuery) DeprecatedField(ctx context.Context) (string, error) {
 }
 func (r *stubQuery) Panics(ctx context.Context) (*Panics, error) {
 	return r.QueryResolver.Panics(ctx)
+}
+func (r *stubQuery) DefaultScalar(ctx context.Context, arg string) (string, error) {
+	return r.QueryResolver.DefaultScalar(ctx, arg)
 }
 func (r *stubQuery) ValidType(ctx context.Context) (*ValidType, error) {
 	return r.QueryResolver.ValidType(ctx)

--- a/plugin/modelgen/models.go
+++ b/plugin/modelgen/models.go
@@ -4,10 +4,9 @@ import (
 	"go/types"
 	"sort"
 
-	"github.com/99designs/gqlgen/internal/code"
-
 	"github.com/99designs/gqlgen/codegen/config"
 	"github.com/99designs/gqlgen/codegen/templates"
+	"github.com/99designs/gqlgen/internal/code"
 	"github.com/99designs/gqlgen/plugin"
 	"github.com/vektah/gqlparser/ast"
 )
@@ -17,6 +16,7 @@ type ModelBuild struct {
 	Interfaces  []*Interface
 	Models      []*Object
 	Enums       []*Enum
+	Scalars     []string
 }
 
 type Interface struct {
@@ -159,6 +159,8 @@ func (m *Plugin) MutateConfig(cfg *config.Config) error {
 			}
 
 			b.Enums = append(b.Enums, it)
+		case ast.Scalar:
+			b.Scalars = append(b.Scalars, schemaType.Name)
 		}
 	}
 
@@ -174,6 +176,9 @@ func (m *Plugin) MutateConfig(cfg *config.Config) error {
 	}
 	for _, it := range b.Interfaces {
 		cfg.Models.Add(it.Name, cfg.Model.ImportPath()+"."+it.Name)
+	}
+	for _, it := range b.Scalars {
+		cfg.Models.Add(it, "github.com/99designs/gqlgen/graphql.String")
 	}
 
 	if len(b.Models) == 0 && len(b.Enums) == 0 {

